lib/RestService/src/RestService.h

```diff
@@ -0,0 +1,312 @@
+
+    /**
+     * Send POST request to host
+     *
+     * @param[in] restId   Unique Id to identify plugin
+     * @param[in] url      URL
+     * @param[in] payload  Payload, which must be kept alive until response is available!
+     * @param[in] size     Payload size in byte
+     *
+     * @return If request is successful sent, it will return true otherwise false.
+     */
+    bool post(const int restId, const String& url, const uint8_t* payload = nullptr, size_t size = 0U);
+
+    /**
+     * Send POST request to host.
+     * @param[in] restId   Unique Id to identify plugin
+     * @param[in] url      URL
+     * @param[in] payload  Payload, which must be kept alive until response is available!
+     *
+     * @return If request is successful sent, it will return true otherwise false.
+     */
+    bool post(const int restId, const String& url, const String& payload);
+
+    /**
+     * Get Response
+     *
+     * @param[in]  restId      Unique Id to identify plugin
+     * @param[out] isValidRsp  Does Response have a payload
+     * @param[out] payload     Content of the Response
+     *
+     * @return If a response is available, it will return true otherwise false
+     */
+
+    bool getResponse(const int restId, bool& isValidRsp, DynamicJsonDocument* payload);
+
+private:
+
+    /**
+     *  Max. number of commands which can be queued. Must be increased when new user of RestService is added
+     */
+    static const size_t CMD_QUEUE_SIZE = 9U;
+
+    struct Msg
+    {
+        int                  restId; /**< Used to identify plugin in RestService */
+        bool                 isMsg;  /**< true: successful Response, false: request failed*/
+        DynamicJsonDocument* rsp;    /**< Response, only valid if isMsg == true */
+
+        /**
+         * Constructs a message.
+         */
+        Msg() :
+            isMsg(false),
+            rsp(nullptr)
+        {
+        }
+    };
+
+    enum CmdId
+    {
+        CMD_ID_GET = 0, /**< GET request */
+        CMD_ID_POST     /**< POST request */
+    };
+
+    /**
+     * A command is a combination of request and its corresponding data.
+     */
+    struct Cmd
+    {
+        CmdId  id;     /**< The command id identifies the kind of request. */
+        int    restId; /**< Used to identify plugin in RestService */
+        String url;    /**< URL */
+
+        /**
+         * The union contains the event id specific parameters.
+         * Note not every command id must have parameters.
+         */
+        union
+        {
+            /**
+             * Data parameters, only valid for CMD_ID_POST.
+             */
+            struct
+            {
+                const uint8_t* data; /**< Command specific data. */
+                size_t         size; /**< Command specific data size in byte. */
+            } data;
+
+        } u;
+    };
+
+    int                    restIdCounter; /**< Used to generate restIds, Increases with each registered plugin */
```

Comment:
Plugin instances can be created and destroyed dynamically. Always just increasing is a good and simple solution, but is not fail safe in theory.